v2/spiffetls/tlsconfig/config.go

```diff
@@ -26,63 +27,86 @@ func HookTLSClientConfig(config *tls.Config, bundle x509bundle.Source, authorize
 	config.VerifyPeerCertificate = WrapVerifyPeerCertificate(config.VerifyPeerCertificate, bundle, authorizer)
 }
 
+// A Option changes the defaults used to by mTLS ClientConfig functions.
+type Option func(*options)
```

Comment:
We should probably use the same pattern we have for other options:

```suggestion
type Option interface {
   apply(*options)
}

type option func(*options)

func (fn option) apply(o *option) {
    fn(o)
}
```

Then the options can be implemented in terms of `option` we get the flexibility later if we do end up needing to have per-method options mixed in with common options.

```
func WithSomeOption() Option {
    return option(func(o *options) {
        // modify `o`
    })
}
```

Reply:
Makes sense, I'd never seen what the purpose of these interface-based options were, so always used the simpler func-based pattern. I see now that this makes it possible to reuse options and still have scoped down option types for narrower use-cases, which obviates the need for the N option types I had included in this PR initially!

Nice! The more you know!